docs/blog/2021-07-30-0.4.0-release.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Recoil 0.4
+---
+
+We are pleased to announce the release of Recoil 0.4 with initial Recoil Transactions, selector cache configuration, and other optimizations and fixes.
+
+## Recoil Transactions
+
+Introducing Recoil Transactions!  The new [`useRecoilTransaction_UNSTABLE()`](/docs/api-reference/core/useRecoilTransaction) API allows for an easier, more performant, and safer way to atomically update multiple atoms together.  For the simple case this could be thought of as similar to the "updater" form of setting Recoil state, but over multiple atoms.  This new hook is easier to use and should replace the need to use the [`useRecoilCallback()`](/docs/api-reference/core/useRecoilCallback) API most of the time.
```

Comment:
We're introducing a [new API] that lets you update multiple atoms together, as a single transaction, in a way that is easier, more efficient, and safer than before. This API replaces most uses of `useRecoilCallback`. However, this release contains only an initial implementation, with certain limitations that will be removed in a future release.